### ISSUE TITLE
tags not saved when creating a project with API

### ISSUE THREAD

**Issue by stboiss** [stevespringett/dependency-track#125]
tags not saved when creating a project with API
```
$ curl -X PUT "http://localhost:8080/api/v1/project" \
>   -H "accept: application/json" \
>   -H "X-Api-Key: UBq4k6VZYejzwPbug8kmd0zU6tNxJEQB" \
>   -H "Content-Type: application/json" \
>   -d "{ \
>       \"name\": \"test\",\
>       \"version\": \"1\",\
>       \"tags\": [ { \"name\": \"test\" } ]\
>       }"
```
results
```
{"name":"test","version":"1","uuid":"5931a119-35bf-4b3a-96f6-a0839d5f5cd1","properties":[],"tags":[]}
```

in Database, in Table "TAG", tag test is created but not linked with project.

**Comment by stevespringett** [comment]
I noticed this as well in the UI. Updating the project seems to work, but the initial creation of the project is where the tags do not seem to be saved.

**Comment by stevespringett** [comment]
A fix for this has been included in the master branch and the 3.0.x branch.
